### ISSUE TITLE
"linux-presto-e2e-tests" failing in GH all the time: "Error occurred in starting fork"

### ISSUE THREAD

**Issue by spershin** [prestodb/presto#21923]
"linux-presto-e2e-tests" failing all the time with "Error occurred in starting fork" in GH.
The test is split into 5 runs to run tests in parallel and one of them is failing pretty much all the time.
Seems always the same: Run 3.

Examples:
https://app.circleci.com/pipelines/github/prestodb/presto/12300/workflows/60ead8ff-e907-49d1-9ff8-8a593df4267e/jobs/47299/parallel-runs/3?filterBy=ALL

https://app.circleci.com/pipelines/github/prestodb/presto/12314/workflows/ec80c75a-c179-4e6f-ba87-134650448674/jobs/47320

Following the link at the end of the logs says:
```
MojoExecutionException
Unlike many other errors, this exception is not generated by the Maven core itself but by a plugin. As a rule of thumb, plugins use this error to signal a problem in their configuration or the information they retrieved from the POM.
```

Observation 1
There are also 615 "Major GC" messages in one log during ~1hr and 95 of these during 7 mins for another PR. Maybe memory use is and issue?

Observation 2
"linux-spark-e2e-tests" is a similar test, whose Run 4 often times out after 5hrs or running. At one instance Run 4 had 2610 events of Major GC and only 6 such events in another run which completed within 5 mins.
The test is very similar to the one mentioned in the title of this issue,
Example: https://app.circleci.com/pipelines/github/prestodb/presto/12300/workflows/6c935db5-1ce9-487a-9361-8314a3abdea6/jobs/47150/parallel-runs/4?filterBy=ALL

Some info from the logs:
```
[INFO] Tests run: 59, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[WARNING] Corrupted channel by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file /root/project/presto-native-execution/target/surefire-reports/2024-02-13T11-56-51_519-jvmRun1.dumpstream
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 53:33 min (Wall Clock)
[INFO] Finished at: 2024-02-13T12:50:21-06:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M7:test (default-test) on project presto-native-execution: 
[ERROR] 
[ERROR] Please refer to /root/project/presto-native-execution/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] ExecutionException The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd '/root/project/presto-native-execution' && '/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.362.b08-3.el8.x86_64/jre/bin/java' '-Dfile.encoding=UTF-8' '-Xmx4g' '-Xms4g' '-XX:+ExitOnOutOfMemoryError' '-XX:+HeapDumpOnOutOfMemoryError' '-XX:-OmitStackTraceInFastThrow' '-jar' '/root/project/presto-native-execution/target/surefire/surefirebooter-20240213115652838_34.jar' '/root/project/presto-native-execution/target/surefire' '2024-02-13T11-56-51_519-jvmRun1' 'surefire-20240213115652838_32tmp' 'surefire_1-20240213115652838_33tmp'
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 3
[ERROR] Crashed tests:
[ERROR] com.facebook.presto.nativeworker.TestPrestoNativeWindowQueries
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: ExecutionException The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd '/root/project/presto-native-execution' && '/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.362.b08-3.el8.x86_64/jre/bin/java' '-Dfile.encoding=UTF-8' '-Xmx4g' '-Xms4g' '-XX:+ExitOnOutOfMemoryError' '-XX:+HeapDumpOnOutOfMemoryError' '-XX:-OmitStackTraceInFastThrow' '-jar' '/root/project/presto-native-execution/target/surefire/surefirebooter-20240213115652838_34.jar' '/root/project/presto-native-execution/target/surefire' '2024-02-13T11-56-51_519-jvmRun1' 'surefire-20240213115652838_32tmp' 'surefire_1-20240213115652838_33tmp'
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 3
[ERROR] Crashed tests:
[ERROR] com.facebook.presto.nativeworker.TestPrestoNativeWindowQueries
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.awaitResultsDone(ForkStarter.java:513)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.runSuitesForkPerTestSet(ForkStarter.java:460)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:327)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:269)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1334)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1167)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:931)
[ERROR]         at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR]         at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:200)
[ERROR]         at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:196)
[ERROR]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[ERROR]         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[ERROR]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[ERROR]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[ERROR]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[ERROR]         at java.lang.Thread.run(Thread.java:750)
[ERROR] Caused by: org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd '/root/project/presto-native-execution' && '/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.362.b08-3.el8.x86_64/jre/bin/java' '-Dfile.encoding=UTF-8' '-Xmx4g' '-Xms4g' '-XX:+ExitOnOutOfMemoryError' '-XX:+HeapDumpOnOutOfMemoryError' '-XX:-OmitStackTraceInFastThrow' '-jar' '/root/project/presto-native-execution/target/surefire/surefirebooter-20240213115652838_34.jar' '/root/project/presto-native-execution/target/surefire' '2024-02-13T11-56-51_519-jvmRun1' 'surefire-20240213115652838_32tmp' 'surefire_1-20240213115652838_33tmp'
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 3
[ERROR] Crashed tests:
[ERROR] com.facebook.presto.nativeworker.TestPrestoNativeWindowQueries
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:714)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.lambda$runSuitesForkPerTestSet$7(ForkStarter.java:449)
[ERROR]         ... 4 more

```

## Expected Behavior
Tests pass or fail in the actual code, not framework.

## Current Behavior
Tests seem to fail in the framework.

## Steps to Reproduce
Submit a PR and observe the test fail.

**Comment by spershin** [comment]
cc @mshang816

**Comment by spershin** [comment]
Observation 3:
For presto e2e we always have "com.facebook.presto.nativeworker.TestPrestoNativeWindowQueries" failing.
For spark e2e the last queries in the log (if you scroll up) have window functions as well:
```
                    min_28 := min(totalprice) ROWS 1 FOLLOWING UNBOUNDED_FOLLOWING (1:574)
                    min_22 := min(totalprice) ROWS 1 PRECEDING 4 PRECEDING (1:438)
```
So far it looks like something is using too much of memory for queries with window functions.

**Comment by spershin** [comment]
The test PR with all window query tests commented out (except the ones in testOverlappingPartitionAndSortingKeys() - missed these and left them uncommented) shows that AbstractTestNativeWindowQueries is indeed responsible for the memory issues that lead to spark test timing out and presto test failing with the weird fork error.
PR: https://github.com/prestodb/presto/pull/21924

Interesting, that there are some legit test failures (https://app.circleci.com/pipelines/github/prestodb/presto?branch=pull%2F21924):

```
TestPrestoNativeWindowQueries>AbstractTestNativeWindowQueries.testOverlappingPartitionAndSortingKeys:202->AbstractTestQueryFramework.assertQuery:149->AbstractTestQueryFramework.assertQuery:154 Execution of 'actual' query failed: SELECT row_number() OVER (PARTITION BY orderdate ORDER BY orderdate) FROM orders
TestPrestoSparkNativeWindowQueries>AbstractTestNativeWindowQueries.testOverlappingPartitionAndSortingKeys:202->AbstractTestQueryFramework.assertQuery:149->AbstractTestQueryFramework.assertQuery:154 Execution of 'actual' query failed: SELECT row_number() OVER (PARTITION BY orderdate ORDER BY orderdate) FROM orders
[ERROR] com.facebook.presto.spark.TestPrestoSparkNativeGeneralQueries.testUnionAllInsert  Time elapsed: 8.84 s  <<< FAILURE!
```

**Comment by mbasmanova** [comment]
@spershin Sergey, thank you for debugging. I'm seeing these failures as well. Let's disable until we figure out how to run them reliably.

**Comment by mbasmanova** [comment]
CC: @majetideepak @tdcmeehan

**Comment by spershin** [comment]
This commit **might** have tipped the test over into the bad zone with the new tests.
https://github.com/prestodb/presto/commit/6eba1f7330de2879ca2f4012c8bbd9b1f7a5f727
Going to try running E2E without these new changes.

cc @aditi-pandit, @pramodsatya

**Comment by spershin** [comment]
PR to back out the 6eba1f7330de2879ca2f4012c8bbd9b1f7a5f727 to allow new PRs to be landed: https://github.com/prestodb/presto/pull/21924

**Comment by jaystarshot** [comment]
Can we also configure manual retry for these tests? Currently they cannot be retried manually

**Comment by spershin** [comment]
> Can we also configure manual retry for these tests? Currently they cannot be retried manually
@jaystarshot

They can be retried manually, that's how folks got by for a while after the tests were added and started to fail the check most of the time. Occasionally checks succeeded.
In order to have retry option enabled you need to be logged in CircleCI in your browser.

**Comment by jaystarshot** [comment]
Ohh thanks

**Comment by jaystarshot** [comment]
When I tried to login it says -
```
 To join your team on CircleCI, contact a teammate and request an invite.
After following the instructions in the invite, you’ll be able to access and contribute to team projects.
Tip: Share the link to [our invite documentation](https://circleci.com/docs/invite-your-team/) with your teammate to help them generate your invite.
```
